www/CQMContext.js

```diff
@@ -311,4 +323,94 @@ CQMContext.prototype._getEntitlementInfo = function(entityNames, successCallback
     exec(successCallback, fail, "CQMContext", "getEntitlementInfo", [entityNames]);
 };
 
+/**
+ * Get status for an Entity such as isSavable, progressPct, progressState. (Current only supports Collection)
+ *
+ * @param {Entity} or {EntityList} Entity we want to get status for or EntityList we want to get status for.
+ * @param {Function} successCallback
+ * @param {Function} errorCallback (OPTIONAL)
+ */
+CQMContext.prototype.getEntityStatus = function(entity, successCallback, errorCallback) {
+
+    var fail = errorCallback && function(code) {
+        var ce = new CQMContextError(code);
+        errorCallback(ce);
+    };
+
+    exec(successCallback, fail, "CQMContext", "getEntityStatus", [entity.type, entity.metadata.entityName]);
+};
+
+/**
+ * Save an Entity.
+ *
+ * @param {Entity} Entity we want to save.
+ * @param {boolean} If we want to save the entity in the background with no notification (Android only)
+ * @param {Function} successCallback that will be called everytime there is a progress update.
+ *                  The callback is given an array that contains two values:
+ *                  1. The latest known entity that was saved
+ *                  2. The download percentage
+ * @param {Function} errorCallback (OPTIONAL)
+ */
+CQMContext.prototype.saveEntity = function(entity, isSilent, successCallback, errorCallback) {
+
+    var success = successCallback && function(json) {
+        var returnEntity = new Entity(json.rawEntity)
+        successCallback([returnEntity, json.downloadPct]);
```

Comment:
Good point. We could return something to the successCallback earlier. Like when an operation was created successfully and give it a 0% to start with. 
Or/And return an extra field, like what we returned for getEntityStatus -- ProgressState.